### PR TITLE
SelectBox: keydown/keyup press should trigger onValueChanged with right e.event when dropDownList is closed (T844170)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/ui/select_box.js [hunk 1]
```diff
@@ -67,18 +67,20 @@ var SelectBox = DropDownList.inherit({
 
                 parent.tab && parent.tab.apply(this, arguments);
             },
-            upArrow: function() {
+            upArrow: function(e) {
                 if(parent.upArrow && parent.upArrow.apply(this, arguments)) {
                     if(!this.option("opened")) {
-                        this._setNextValue(-1);
+                        e.step = -1;
```
Comment (San4es) [inline js/ui/select_box.js]: Can we avoid this?

Comment (ksercs) [reply js/ui/select_box.js]: yeah, ready

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -3802,6 +3802,24 @@ QUnit.module("keyboard navigation", moduleSetup, () => {
         assert.strictEqual(instance.option("value"), 1, "upArrow");
     });
 
+    ["ArrowDown", "ArrowUp"].forEach((key) => {
+        QUnit.test(`${key}} should trigger onValueChanged with right e.event when dropDown is closed (T844170)`, function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]:
```suggestion
        QUnit.test(`${key} should trigger onValueChanged with right e.event when dropDown is closed (T844170)`, function(assert) {
```

**Review by San4es — DISMISSED** [review]
LGTM

**Review by San4es — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**